tests/integration_tests/test_visualisations.py

```diff
@@ -668,6 +782,72 @@ def test_visualisation_compare_classifiers_predictions_output_saved(
             print("Error: %s - %s." % (e.filename, e.strerror))
 
 
+def test_visualisation_compare_classifiers_predictions_csv_output_saved(
+        csv_filename
+):
+    """Ensure pdf and png figures from the experiments can be saved.
+
+    Predictions are loaded form csv file.
+    :param csv_filename: csv fixture from tests.fixtures.filenames.csv_filename
+    :return: None
+    """
+    input_features = [
+        text_feature(vocab_size=10, min_len=1, representation='sparse'),
+        categorical_feature(
+            vocab_size=10,
+            loss='sampled_softmax_cross_entropy'
+        )
+    ]
+    output_features = [categorical_feature(vocab_size=2, reduce_input='sum')]
+
+    # Generate test data
+    rel_path = generate_data(input_features, output_features, csv_filename)
+    encoder = 'cnnrnn'
+    input_features[0]['encoder'] = encoder
+    exp_dir_name = run_experiment(input_features, output_features,
+                                  data_csv=rel_path)
+    vis_output_pattern_pdf = exp_dir_name + '/*.pdf'
+    vis_output_pattern_png = exp_dir_name + '/*.png'
+    field_name = get_output_field_name(exp_dir_name)
+    prediction = exp_dir_name + '/{}_predictions.csv'.format(field_name)
+    experiment_source_data_name = csv_filename.split('.')[0]
+    ground_truth = experiment_source_data_name + '.hdf5'
+    test_cmd_pdf = ['python',
+                    '-m',
+                    'ludwig.visualize',
+                    '--visualization',
+                    'compare_classifiers_predictions',
+                    '--ground_truth',
+                    ground_truth,
+                    '--field',
+                    field_name,
+                    '--predictions',
+                    prediction,
+                    prediction,
+                    '--model_names',
+                    'Model1',
+                    'Model2',
+                    '-od', exp_dir_name]
+    test_cmd_png = test_cmd_pdf.copy() + ['-ff', 'png']
+
+    commands = [test_cmd_pdf, test_cmd_png]
+    vis_patterns = [vis_output_pattern_pdf, vis_output_pattern_png]
+
+    for command, viz_pattern in zip(commands, vis_patterns):
+        result = subprocess.run(command)
+        figure_cnt = glob.glob(viz_pattern)
+
+        assert 0 == result.returncode
+        assert 1 == len(figure_cnt)
+
+    shutil.rmtree(exp_dir_name, ignore_errors=True)
+    shutil.rmtree('results', ignore_errors=True)
+    for file in glob.glob(experiment_source_data_name + '.*'):
+        try:
+            os.remove(file)
+        except OSError as e:  # if failed, report it back to the user
+            print("Error: %s - %s." % (e.filename, e.strerror))
```

Comment:
Maybe add a little more info here, Error deleting file or something like that?

Reply:
The e.strerror carries the original OSError exception message. I think thtat should be sufficient information on what went wrong.